### PR TITLE
Add support for NetworkPolicy.

### PR DESCRIPTION
**Pull request by desaintmartin**
NetworkPolicy support would be a nice addition to this chart. Here is a implementation proposal, open to discussion, where I strictly separate http and transport ports.

- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [x] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Comment by desaintmartin** [comment]
Up?

**Comment by Mykolaichenko** [comment]
Upvoting for NetworkPolicy, @desaintmartin thank you :)

**Comment by desaintmartin** [comment]
I can rebase if you are interested into this feature. ;)

**Comment by desaintmartin** [comment]
Not stale... Is this PR wanted?

**Comment by desaintmartin** [comment]
cc @jmlrt maybe ?

**Comment by jmlrt** [comment]
Hi @desaintmartin, thanks for this PR. This is still something that we want to merge but don't have workload for a proper review and tests currently.

File: elasticsearch/examples/networkpolicy/test/goss.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+kernel-param:
```
Comment (jmlrt) [inline elasticsearch/examples/networkpolicy/test/goss.yaml:1]: If there is no specific test relative to network policies, the `test` target and `goss.yaml` isn't necessary for this example.

File: elasticsearch/examples/networkpolicy/Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+default: test
+include ../../../helpers/examples.mk
+
+RELEASE := helm-es-networkpolicy
+
+install:
+	helm upgrade --wait --timeout=600 --install $(RELEASE) --values ./values.yaml ../../ ; \
```
Comment (jmlrt) [inline elasticsearch/examples/networkpolicy/Makefile:7]: timeout option now requires `s` for second with Helm 3

**Review by jmlrt — CHANGES_REQUESTED** [review]
Hi @desaintmartin, sorry for the delay, I finally was able to review it.

This is mostly LGTM. 

- Can you merge rebase on master to fix conflicts?
- The `networkpolicy` example may not be required if there is no tests specifics to network policies. we could remove the `goss.yaml`, but if the examples values are the same as commented example in default values, we may not need this example at all
- Finally the python tests may need to be formatted using [Black](https://black.readthedocs.io/en/stable/) to pass CI tests

File: elasticsearch/examples/networkpolicy/Makefile (hunk 2 above)
Comment (desaintmartin) [reply elasticsearch/examples/networkpolicy/Makefile:7]: Updated!

File: elasticsearch/examples/networkpolicy/test/goss.yaml (hunk 1 above)
Comment (desaintmartin) [reply elasticsearch/examples/networkpolicy/test/goss.yaml:1]: Should it be something like
```
test: install
	helm test $(RELEASE)
```

then?

**Comment by desaintmartin** [comment]
Done! Let's see what the CI says.

**Comment by jmlrt** [comment]
jenkins test this please

File: elasticsearch/tests/elasticsearch_test.py [hunk 3]
```diff
@@ -1367,3 +1367,101 @@ def test_hostaliases():
...
+            }
+        },
+        {"podSelector": {"matchLabels": {"app": "elasticsearch-master"}}},
+    ]
+    assert transport["ports"][0]["port"] == 9300
+    assert pod_selector == {
+        "matchLabels": {
+            "app": "elasticsearch-master",
+        }
+    }
```
Comment (jmlrt) [inline elasticsearch/tests/elasticsearch_test.py:1467]: This part need reformating to make https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+lint-python/label=docker&&virtual&&ubuntu-18.04/745/console pass

File: elasticsearch/examples/networkpolicy/Makefile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+install:
+	helm upgrade --wait --timeout=600s --install $(RELEASE) --values ./values.yaml ../../ ; \
+
+restart:
+	helm upgrade --set terminationGracePeriod=121 --wait --timeout=600s --install $(RELEASE) --values ./values.yaml ../../ ; \
+
+test: install
+	helm test $(RELEASE)
+
```
Comment (jmlrt) [inline elasticsearch/examples/networkpolicy/Makefile:14]: >Should it be something like
>```
>test: install
>	helm test $(RELEASE)
>```

I would suggest dropping completely the test target if there is no specific tests:

```suggestion
```

**Review by jmlrt — CHANGES_REQUESTED** [review]

File: elasticsearch/tests/elasticsearch_test.py (hunk 3 above)
Comment (desaintmartin) [reply elasticsearch/tests/elasticsearch_test.py]: Funny that my black tries to reformat against the one from CI, there must be some mismatch in their configuration. Manually fixed.

**Comment by desaintmartin** [comment]
Updated!

**Comment by jmlrt** [comment]
jenkins test this please

**Review by jmlrt — APPROVED** [review]
LGTM⛴

**Pull request merged**